### PR TITLE
fix calling js hostFunction from worklet runtime

### PR DESCRIPTION
**Pull request by Szymon20000**

### CODE REVIEW

File: cpp/JsiWorkletContext.cpp [hunk 1]
```diff
@@ -352,17 +344,23 @@ JsiWorkletContext::createCallInContext(jsi::Runtime &runtime,
...
-              auto results = workletInvoker->call(
-                  runtime, thisWrapper->unwrap(runtime),
-                  ArgumentsWrapper::toArgs(args), argsWrapper.getCount());
+              jsi::Value result;
+              if (workletInvoker != nullptr) {
+                result = workletInvoker->call(
+                      runtime, thisWrapper->unwrap(runtime),
+                      ArgumentsWrapper::toArgs(args), argsWrapper.getCount());
+              } else {
+                result = func->call(
```
[the remark this answers is not in the document]
Comment (Szymon20000) [reply cpp/JsiWorkletContext.cpp:358]: Sure!

let me explain what's currently happening in Wishlist without that change.
We have js function that is not a worklet. We call it from worklet thread.
Here is how we create it on JS thread!
```
const attachGestureHandler = createRunInJsFn((tag: number) => {
```
Then we want to call it from a worklet thread. However, as it's not a worklet workletInvoker is nullptr and calling `attachGestureHandler` results in crash.

[the remark this answers is not in the document]
Comment (Szymon20000) [reply cpp/JsiWorkletContext.cpp:358]:
```
export const createRunInJsFn = Worklets.createRunInJsFn;
```

[the remark this answers is not in the document]
Comment (Szymon20000) [reply cpp/JsiWorkletContext.cpp:358]: If it's crashing in other cases then maybe we can just add the following check?
```c++
 // Ensure that the function is a worklet
    if (workletInvoker == nullptr && convention != CallingConvention::CtxToJs) {
      throw jsi::JSError(runtime, "In callInContext the function parameter "
                                  "is not a valid worklet and "
                                  "cannot be called between contexts or "
                                  "from/to JS from/to a context.");
    }
```

[the remark this answers is not in the document]
Comment (Szymon20000) [reply cpp/JsiWorkletContext.cpp:358]: After I added that is still works for our case as our calling convention is `CallingConvention::CtxToJs`

**Review by chrfalch — APPROVED** [review]

**Pull request merged**